### ISSUE TITLE
Table of contents cannot track headers with Raw Block

### ISSUE THREAD

**Issue by gha88** [TheRenegadeCoder/SnakeMD#161]
As expected table of contents is rendered only before printing the document.
When rendering it is populated searching fow Heders block in document.
The problem is that using Raw block some headers can be set inside.
In my opinion the headers content should be track anyway even if inside the raw content.
The problem is that the content in Raw block should be parsed somehow.

**Comment by jrg94** [comment]
Yeah, that makes a lot of sense. At the moment, the raw blocks aren't parsed. This is something I wanted to implement for a while (see https://github.com/TheRenegadeCoder/SnakeMD/issues/53 and https://github.com/TheRenegadeCoder/SnakeMD/issues/1). However, I don't know how seen that would possibly be without some help.

**Comment by gha88** [comment]
How would you like to implement it?
Are you thinking to embed an external parsing library (creating a dependency) or to create a minimal internal parser?

Would you parse to fix this issue only or do you have some other issues or cases to parse on?

**Comment by jrg94** [comment]
Ideally, I would like to avoid adding external libraries altogether. That said, I've recently dabbled with templates in the library, and I think it would be possible to implement a 3rd-party template with external dependencies. In other words, someone could make their own library that could be imported separately as an extension of SnakeMD. I've written a little about how that would work [here](https://www.snakemd.io/en/latest/docs/template-api/#templates). 

Alternatively, we could write a parser in SnakeMD to parse markdown into Elements. The original idea for that was to be able to parse raw blocks, so we could import existing markdown files (e.g., headers, footers, etc.). I think it would have a lot of use cases regardless. 

A wild third option would be to create an alternative implementation of the Table of Contents objects, which just scans the rendered document for headings on each line (`#`, `##`, ...). There could even be an optional parameter for this on the existing table of contents, which turns on a parsing mode.

**Comment by gha88** [comment]
In my specific case, to solve the problem, I'm going to develop a custom renderer from [mistletoe](https://github.com/miyuchina/mistletoe/tree/master) library that I'm already using in the project.

The idea is to inject the SnakeMd document in the custom renderer. 
Make a custom renderer in this way should be far less difficult than create it from zero.

I think the raw block in snake md is good as is because for example you can use it to avoid some headers from the table of contents (the opposite of my problem).

I don't know if you could be interested in someway about this custom renderer. Maybe it can be considered as a plugin or something similar.

In that case let me know.
